libraries/AP_GPS/AP_GPS_NOVA.cpp

```diff
@@ -276,10 +315,55 @@ AP_GPS_NOVA::process_message(void)
         const psrdop &psrdopu = nova_msg.data.psrdopu;
 
         state.hdop = (uint16_t) (psrdopu.hdop*100);
-        state.vdop = (uint16_t) (psrdopu.htdop*100);
+        //state.vdop = (uint16_t) (psrdopu.htdop*100);
+        state.vdop = (uint16_t)((sqrt(psrdopu.pdop - psrdopu.hdop)) * 100);
         return false;
     }
 
+    if(messageid == 971 || messageid == 2042)
+    {
+        const nova_heading &headingu = nova_msg.data.headingu;
+        if (headingu.solstat == 0) {// have a solution
+            switch (headingu.postype)
+            {
+                case 16:
+                case 17: // psrdiff
+                case 18: // waas
+                case 20: // omnistar
+                case 68: // ppp_converg
+                case 69: // ppp
+                case 32: // l1 float
+                case 33: // iono float
+                case 34: // narrow float
+                case 48: // l1 int
+                case 50: // narrow int
+                {
+                    state.have_gps_yaw = true;
+                    state.gps_yaw_configured = true;
+                    state.have_gps_yaw_accuracy = true;
+                    state.gps_yaw = (float)(headingu.heading);
+                    state.gps_yaw_accuracy = (float)(headingu.hdgstddev);
+                    state.gps_yaw_time_ms = AP_HAL::millis();
+                    break;
+                }
+                case 0: // NONE
+                case 1: // FIXEDPOS
+                case 2: // FIXEDHEIGHT
+                default:
+                {
+                    state.have_gps_yaw = false;
+                    state.gps_yaw_configured = false;
+                    state.have_gps_yaw_accuracy = false;
+                    break;
+                }
+            }
+        } else {
```

Comment:
this could lead to oscillating have_gps_yaw depending on what messages are sent. Maybe better to disable GPS yaw if gps_yaw_time_ms is older than 1 second at the end of AP_GPS_NOVA::read() ?